### Question 1

Run docker with the python:3.13 image. Use an entrypoint bash to interact with the container.

What's the version of pip in the image?

In [4]:
!docker run -it --entrypoint bash python:3.13

root@7bf19bb5ff34:/# ^C

root@7bf19bb5ff34:/# 

In [5]:
!docker run --rm python:3.13 pip --version

pip 25.3 from /usr/local/lib/python3.13/site-packages/pip (python 3.13)


Answer: 25.3

### Question 2

Given the following docker-compose.yaml, what is the hostname and port that pgadmin should use to connect to the postgres database?

services:
  db:
    container_name: postgres
    image: postgres:17-alpine
    environment:
      POSTGRES_USER: 'postgres'
      POSTGRES_PASSWORD: 'postgres'
      POSTGRES_DB: 'ny_taxi'
    ports:
      - '5433:5432'
    volumes:
      - vol-pgdata:/var/lib/postgresql/data

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4:latest
    environment:
      PGADMIN_DEFAULT_EMAIL: "pgadmin@pgadmin.com"
      PGADMIN_DEFAULT_PASSWORD: "pgadmin"
    ports:
      - "8080:80"
    volumes:
      - vol-pgadmin_data:/var/lib/pgadmin

volumes:
  vol-pgdata:
    name: vol-pgdata
  vol-pgadmin_data:
    name: vol-pgadmin_data

The answers are both of these:
1. `postgres:5432`
2. `db:5432`

The service name is `db` and the container name is `postgres`, so they both point to the same container. The important part is that it listens on port 5432.

### Question 3